airbyte-commons-worker/src/main/java/io/airbyte/workers/helper/CatalogClientConverters.java

```diff
@@ -17,6 +28,86 @@
  */
 public class CatalogClientConverters {
 
+  /**
+   *
+   * @param catalog
+   * @return
+   */
+  public static io.airbyte.protocol.models.AirbyteCatalog toAirbyteProtocol(
```

Comment:
It definitely makes me feel uneasy to duplicate such complicated and hard-to-maintain code. It seems like it will be very likely that this code will need to change in both places, but it will only end up being changed in a single place because it isn't obvious that there are multiple implementations of nearly the exact same thing.

I don't really have a good answer for this right now, since the conversion logic isn't actually duplicated (in one case, they are server-side java models, and here they are client-side java models, so they're technically different types).

I think it would be ideal to add very apparent comments to this class AND to the other CatalogConverter class that the logic is duplicated for now, and we should discuss and try to prioritize an issue for coming up with a better approach for these super-complicated catalog conversion methods. 

I don't want to block this PR, but I want to make sure we're doing everything we can to minimize the tech debt here and come up with a solid long term solution!

Reply:
I agree - conversion between those three types with added logic for stream makes it really hard to maintain.